datus-hive/datus_hive/connector.py

```
@@ -262,21 +264,24 @@ def get_sample_rows(
-                        "identifier": self.identifier(database_name=database_name, table_name=table_name),
-                        "catalog_name": "",
-                        "database_name": database_name,
-                        "schema_name": "",
-                        "table_name": table_name,
-                        "table_type": table_type,
-                        "sample_rows": df.to_csv(index=False),
-                    }
-                )
+            try:
+                full_name = self.full_name(database_name=database_name, table_name=table_name)
+                query = f"SELECT * FROM {full_name} LIMIT {top_n}"
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)